### PR TITLE
Fix breakages in AmpContext due to use of IframeMessagingClient

### PR DESCRIPTION
**Pull request by bradfrizzell**
Found some mistakes/bugs in AmpContext due to changes to using the IframeMessagingClient. Fixed them.

### CODE REVIEW

File: 3p/ampcontext.js [hunk 1]
```diff
@@ -109,7 +109,7 @@ export class AmpContext {
    */
   setupMetadata_() {
     try {
-      const data = JSON.parse(decodeURI(this.win_.name));
+      const data = JSON.parse(decodeURI(this.client_.win_.name));
```
Comment (lannka) [inline 3p/ampcontext.js]: should not to access private member.

**Comment by bradfrizzell** [comment]
I can't really add a test until the last PR (the sentinel refactor #6825) is through, because without that AmpContext doesn't work anyway, it's basically just a file in isolation right now. I have a tests PR that I'll submit once the sentinel refactor is through. Is that alright?

File: src/iframe-helper.js [hunk 2]
```diff
@@ -381,10 +382,10 @@ function getSentinel_(iframe, opt_is3P) {
  */
 function parseIfNeeded(data) {
   const shouldBeParsed = typeof data === 'string'
-      && data.charAt(0) === '{';
+      && data.indexOf('amp-') == 0;
```
Comment (lannka) [inline src/iframe-helper.js]: Is this changed needed right now? No code except IframeMessagingClient is sending message in this format yet. I would expect a deprecation of this whole method in your refactoring CL.

File: 3p/ampcontext.js [hunk 3]
```diff
@@ -109,7 +109,7 @@ export class AmpContext {
    */
   setupMetadata_() {
     try {
-      const data = JSON.parse(decodeURI(this.win_.name));
+      const data = JSON.parse(decodeURI(this.client_.getWindow().name));
```
Comment (lannka) [inline 3p/ampcontext.js]: I think we can just save win as a local field in constructor instead of introducing a dependency.

Comment (bradfrizzell) [inline 3p/ampcontext.js]: sorry, do you mean a public property on iframemessagingclient, another private property on ampcontext, or something else?

File: src/iframe-helper.js (hunk 2 above)
Comment (bradfrizzell) [inline src/iframe-helper.js]: ok, I can move this change.

File: 3p/ampcontext.js [hunk 4]
```diff
@@ -99,7 +99,7 @@ export class AmpContext {
    *  @param {HTMLIframeElement} iframe The iframe we are adding the context to.
    */
   addContextToIframe(iframe) {
-    iframe.name = this.win_.name;
+    iframe.name = this.client.getWindow().name;
```
Comment (jridgewell) [inline 3p/ampcontext.js]: `client_`. Are there tests for `addContextToIframe`?

**Pull request closed without merging**